src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -255,6 +226,59 @@ public virtual PipeTransmissionMode ReadMode
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        private static string s_pipeDirectoryPath;
+
+        private static string GetPipeDirectoryPath()
+        {
+            string path = s_pipeDirectoryPath;
+            if (path == null)
+            {
+                // Create the pipes temporary directory, e.g. "/tmp/.dotnet/corefx/pipes".
+                // We don't have access to Directory.CreateDirectory (which handles creating
+                // all levels of the directory full path specified), so we create each nested
+                // level individually (though we assume that the temp path already exists).
+
+                path = Path.GetTempPath();
+
+                path = Path.Combine(path, PersistedFiles.TopLevelHiddenDirectory);
+                CreateDirectory(path);
+
+                path = Path.Combine(path, PersistedFiles.SecondLevelDirectory);
+                CreateDirectory(path);
+
```

Comment:
CreateDirectory should create all intermediate directories it needs, and Path.Combine doesn't care about existence...

But also a "why isn't this using a method off of PersistedFiles"?  (I like the idea of the directory names staying private, vs internal; maybe it's just me)